Util/src/main/java/io/deephaven/util/codec/SimpleByteArrayCodec.java

```diff
@@ -0,0 +1,77 @@
+package io.deephaven.util.codec;
+
+import io.deephaven.datastructures.util.CollectionUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class SimpleByteArrayCodec implements ObjectCodec<byte[]> {
+    private final int expectedWidth;
+
+    public SimpleByteArrayCodec(@Nullable final String arguments) {
+        if (arguments == null || arguments.trim().isEmpty()) {
+            expectedWidth = ObjectCodec.VARIABLE_WIDTH_SENTINEL;
+            return;
+        }
+        final int size;
+        final String[] tokens = arguments.split(",");
+        if (tokens.length == 0) {
+            expectedWidth = ObjectDecoder.VARIABLE_WIDTH_SENTINEL;
+            return;
+        }
+        try {
+            size = Integer.parseInt(tokens[0].trim());
+            if (tokens.length > 1) {
+                throw new IllegalArgumentException("Unexpected additional arguments after first: " + arguments);
+            }
+        } catch (NumberFormatException ex) {
+            throw new IllegalArgumentException("Error parsing column size: " + ex.getMessage(), ex);
+        }
+        if (size < 1) {
+            throw new IllegalArgumentException("Invalid column size: " + size);
+        }
+        expectedWidth = size;
+    }
+
+    @NotNull
+    @Override
+    public byte[] encode(@Nullable final byte[] input) {
+        if (input == null || input.length == 0) {
+            return CollectionUtil.ZERO_LENGTH_BYTE_ARRAY;
+        }
+        return input;
+    }
+
+    @Override
+    public boolean isNullable() {
+        return true;
+    }
+
+    @Override
+    public int getPrecision() {
+        return 0;
+    }
+
+    @Override
+    public int getScale() {
+        return 0;
+    }
+
+    @Nullable
+    @Override
+    public byte[] decode(@NotNull final byte[] input, final int offset, final int length) {
+        if (input.length == 0) {
+            return null;
+        }
+        if (offset == 0 && length == input.length) {
+            return input;
+        }
+        final byte[] output = new byte[length];
+        System.arraycopy(input, offset, output, 0, length);
```

Comment:
Note that if `length > input.length - offset` we're going to throw an AIOOBE. I think that's OK, and I'm not advocating a change for that.

Reply:
Good point; I thought maybe we could wrap that failure as an IllegalStateException by catching and rethrowing with the AIOOBE wrapped, but then again, probably too much, enough information as it is without nesting dolls.